mod.ts

```diff
@@ -57,11 +57,20 @@ export function evaluate<T>(block: () => Computation, getNext = $next()): T {
 
 function oneshot<T, R>(fn: Continuation<T, R>): Continuation<T, R> {
   let continued = false;
+  let failure: { error: unknown };
   let result: any;
+
   return ((value) => {
     if (!continued) {
       continued = true;
-      return result = fn(value);
+      try {
+        return result = fn(value);
+      } catch (error) {
+        failure = { error };
+        throw error;
+      }
+    } else if (failure) {
+      throw failure.error;
```

Comment:
Does this mean that if `k()` was called twice it would always throw the same error?

Reply:
Exactly 👍🏻

Reply:
I had to wrap it in an object because technically in JS `false`, `0`, `""`, `undefined`, and `null` along with anything else falsey are totally valid things to throw (although anybody who does that should be put into a rocket ship and fired into the sun)